src/axom/primal/geometry/Cone.hpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_PRIMAL_CONE_HPP_
+#define AXOM_PRIMAL_CONE_HPP_
+
+#include "axom/core.hpp"
+
+#include "axom/primal/constants.hpp"
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Vector.hpp"
+
+#include "axom/slic/interface/slic.hpp"
+#include "axom/fmt.hpp"
+
+#include <ostream>
+
+namespace axom
+{
+namespace primal
+{
+/*!
+ * \class Cone
+ *
+ * \brief Represents a cone defined by a base radius,
+ * a top radius, the length from base to top along its axis,
+ * the orientation of the axis and a translation.
+ * \tparam T the coordinate type, e.g., double, float, etc.
+ * \tparam NDIMS the number of spatial dimensions
+ *
+ * A negative length is allowed and results in a negative volume.
+ * Radii must be non-negative.
+ *
+ * A cylinder can be represented using equal base and top radii.
+ */
+template <typename T, int NDIMS>
+class Cone
+{
+public:
+  using PointType = Point<T, NDIMS>;
+  using VectorType = Vector<T, NDIMS>;
+
+public:
+  /*!
+   * \brief Default constructor constructs a degenerate cone,
+   * oriented along the first axis.
+   */
+  AXOM_HOST_DEVICE Cone()
+    : m_baseZ(0.0)
+    , m_baseRad(0.0)
```

Comment:
Would it make more sense to set m_baseRad = 1. and m_topZ = 1. so the cone is not degenerate by default? Or, is this used elsewhere in the logic for Cone?

Reply:
It's not currently used in the clipping code, where the need for `Cone` arose.  I'm not sure what the default should be.  We do it both ways in existing geometries.  Most `primal` geometries default to a degenerate geometry but `Sphere` defaults to a unit sphere.